pkg/vpc/cleanup.go

```diff
@@ -17,7 +20,7 @@ func fmtSecurityGroupNameRegexForCluster(name string) string {
 	return fmt.Sprintf(ourSecurityGroupNameRegexFmt, name)
 }
 
-func findAvailableNetworkInterfaces(provider api.ClusterProvider, spec *api.ClusterConfig) ([]string, error) {
+func findAvailableNetworkInterfaces(ctx context.Context, ec2API ec2iface.EC2API, spec *api.ClusterConfig, fn func(eniID string) error) error {
```

Comment:
This goes through all the network interfaces, right? Is so, I would rename things like this:

```suggestion
func visitAvailableNetworkInterfaces(ctx context.Context, ec2API ec2iface.EC2API, spec *api.ClusterConfig, visitor func(eniID string) error) error {
```

Reply:
It goes through available network interfaces belonging to the cluster's VPC. `visit*` to me sounds more like something used to traverse a tree or graph, and `visitor` (non-verb) as a function name sounds like an interface that'd have a `Visit` method. By convention, most libraries, including the standard library, use just `fn` to denote a callback function.
`forEach*`, IMO, may be more appropriate here.

Also, @errordeveloper what do you think about using *DanglingNetworkInterfaces instead of *AvailableNetworkInterfaces?